api/src/auth/auth.ts

```diff
@@ -44,7 +44,7 @@ export default abstract class Auth {
 	 * @param user User information
 	 * @throws InvalidCredentialsException
 	 */
-	refresh(_user: User): void {
+	async refresh(_user: User): Promise<void> {
```

Comment:
I prefer not enforcing async. If the inheriting class doesn't need the method to be to be async it shouldn't have to define it. Mostly for understandably. If I saw a function implementing async without using an await, I would assume someone forgot their await.